### ISSUE TITLE
Type-casting methods should return the exact type

### ISSUE THREAD

**Issue by senese** [sloria/environs#367]
Using environs I noticed that type-casting methods they don't return the type casted, only `Unknown`,

![image](https://github.com/user-attachments/assets/c0e633bc-2042-4faf-9a3c-91e28a544b40)

If the intention is to cast the environment variable to the correct type it should let your IDE know that it is indeed that type being returned. Environs should leverage `marshmallow` to return the exact type that is being casted. Type checkers like mypy would benefit from this behavior as well.

One more thing is that type-casting methods are not exactly methods and are being treated as variables although the syntax used is like calling functions. Maybe this should be reviewed in the next version?

**Comment by sloria** [comment]
I agree that this would be a nice improvement. Would certainly appreciate a PR (even if just a POC) for this
